docs/developer/decision-records/2022-10-10-naming-conventions/README.md

```diff
@@ -0,0 +1,122 @@
+# EDC Naming Conventions
+
+## Decision
+
+The naming of existing and future Java packages, Gradle modules, Maven artifacts, and configuration
+properties should follow defined naming conventions that align the project's structure. 
+
+Related decision records:
+- [2022-08-09 Project structure review](../2022-08-09-project-structure-review/)
+- [2022-08-11 Versioning and Artifacts](../2022-08-11-versioning_and_artifacts/)
+
+## Rationale
+
+A software project's structure should be designed as developer-friendly as possible, by following precisely
+defined rules based on established standards.
+
+Our goals by introducing naming conventions are:
+- Easy navigation of either Gradle modules, Maven artifacts or the Java package structure.
+- Unique naming of config properties (especially with regard to custom properties introduced by extensions).
+- Accordance to the Eclipse Foundation's guidelines and release processes.
+- Elimination of split packages.
+
+## Approach
+
+### Gradle module name
+
+Gradle modules must have unique names across the entire build classpath, even if they are located in
+different module paths. This is because of a bug in Gradle itself where Gradle will erroneously report
+a "cyclic dependency" if this rule is violated. The following hypothetical example would constitute
+such a violation:
+
+```kotlin
+// settings.gradle.kts
+include(":core:common:transfer-process")
+include(":extensions:sql:transfer-process")
+```
+
+The EDC project has checks in place to make sure module IDs are unique.
+
+> Rule 1: Modules must have unique names.
+
+In addition, the _module name_ should give a hint what is in the module, without being too verbose. The
+earlier example would be a bad one, because "transfer-process" does not indicate what the contents could
+be. This is especially important because Maven's _artifactId_ must be equal to module names.
```

Comment:
```suggestion
be. This is especially important because we require the Maven's _artifactId_ to be equal to module names.
```
That makes it clearer that it is not actually a requirement imposed by Maven or Gradle